src/fides/api/ops/models/privacy_notice.py

```diff
@@ -0,0 +1,191 @@
+
+
+class ConsentMechanism(Enum):
+    opt_in = "opt_in"
+    opt_out = "opt_out"
+    necessary = "necessary"
+
+
+class EnforcementLevel(Enum):
+    """
+    Enum is not formalized in the DB because it may be subject to frequent change
+    """
+
+    frontend = "frontend"
+    system_wide = "system_wide"
+
+
+class PrivacyNoticeBase:
+    """
+    Base class to establish the common columns for `PrivacyNotice`s and `PrivacyNoticeHistory`s
+    """
+
+    name = Column(String, nullable=False)
+    description = Column(String, nullable=True)
+    origin = Column(String, nullable=True)  # pointer back to an origin template ID
+    regions = Column(
+        ARRAY(EnumColumn(PrivacyNoticeRegion, native_enum=False)),
+        index=True,
+        nullable=False,
+    )
+    consent_mechanism = Column(EnumColumn(ConsentMechanism), nullable=False)
+    data_uses = Column(
+        ARRAY(String), nullable=False
+    )  # a list of `fides_key`s of `DataUse` records
+    enforcement_level = Column(EnumColumn(EnforcementLevel), nullable=False)
+    version = Column(Float, nullable=False, default=1.0)
+    disabled = Column(Boolean, nullable=False, default=False)
+    has_gpc_flag = Column(Boolean, nullable=False, default=False)
+    displayed_in_privacy_center = Column(Boolean, nullable=False, default=True)
+    displayed_in_banner = Column(Boolean, nullable=False, default=True)
+    displayed_in_privacy_modal = Column(Boolean, nullable=False, default=True)
+
+
+class PrivacyNotice(PrivacyNoticeBase, Base):
+    """
+    A notice set up by a system administrator that an end user (i.e., data subject)
+    accepts or rejects to indicate their consent for particular data uses
+    """
+
+    def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyNotice:
+        """
+        Overrides the base update method to automatically bump the version of the
+        PrivacyNotice record and also create a new PrivacyNoticeHistory entry
+        """
+        # history record data is identical to privacy notice record data
+        # except the notice's 'id' must be moved to the FK column
+        # and is no longer the history record 'id' column
+        history_data = {
+            "name": self.name,
+            "description": self.description or None,
+            "origin": self.origin or None,
+            "regions": self.regions,
+            "consent_mechanism": self.consent_mechanism,
+            "data_uses": self.data_uses,
+            "enforcement_level": self.enforcement_level,
+            "version": self.version,
+            "disabled": self.disabled,
+            "has_gpc_flag": self.has_gpc_flag,
+            "displayed_in_privacy_center": self.displayed_in_privacy_center,
+            "displayed_in_banner": self.displayed_in_banner,
+            "displayed_in_privacy_modal": self.displayed_in_privacy_modal,
+            "privacy_notice_id": self.id,
+        }
+        PrivacyNoticeHistory.create(db, data=history_data, check_name=False)
```

Comment:
interesting, that does seem like a reasonable thing to consider, and i don't really see any immediate downsides. do you remember why we moved away from that? i don't think i was involved at that point, but maybe i'm just forgetting 😄

Reply:
Ah, this was just how I'd laid it out in the design doc, I had planned to have the PrivacyNoticeHistory table have every version and you'd gone a different direction here, which is definitely fine, I didn't have a very strong opinion here.  I assumed you had reasons for implementing it differently.

Reply:
oh i see - i'd missed that, whoops! i do think it's probably more comprehensive/cleaner. now that i'm getting some more test coverage in place, i'd feel a bit better about making this update rather "last minute", so let me give it a go.

agreed it's probably not a _must have_, but i do like it better, so it feels like better to do it now, earlier rather than later.

Reply:
ah this looks good